x-pack/plugins/security_solution/server/endpoint/utils/create_internal_readonly_so_client.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest, SavedObjectsClientContract, SavedObjectsServiceStart } from 'kibana/server';
+import { EndpointError } from '../errors';
+
+type SavedObjectsClientContractKeys = keyof SavedObjectsClientContract;
+
+const RESTRICTED_METHODS: readonly SavedObjectsClientContractKeys[] = [
+  'bulkCreate',
+  'bulkUpdate',
+  'create',
+  'createPointInTimeFinder',
+  'delete',
+  'removeReferencesTo',
+  'openPointInTimeForType',
+  'closePointInTime',
+  'update',
+  'updateObjectsSpaces',
+];
+
+export class InternalReadonlySoClientMethodNotAllowedError extends EndpointError {}
+
+/**
+ * Creates an internal (system user) Saved Objects client (permissions turned off) that can only perform READ
+ * operations.
+ */
+export const createInternalReadonlySoClient = (
+  savedObjectsServiceStart: SavedObjectsServiceStart
+): SavedObjectsClientContract => {
+  const fakeRequest = ({
+    headers: {},
+    getBasePath: () => '',
+    path: '/',
+    route: { settings: {} },
+    url: { href: {} },
+    raw: { req: { url: '/' } },
+  } as unknown) as KibanaRequest;
+
+  const internalSoClient = savedObjectsServiceStart.getScopedClient(fakeRequest, {
+    excludedWrappers: ['security'],
+  });
+
+  return new Proxy(internalSoClient, {
+    get(
+      target: SavedObjectsClientContract,
+      methodName: SavedObjectsClientContractKeys,
+      receiver: unknown
+    ): unknown {
+      if (RESTRICTED_METHODS.includes(methodName)) {
```

Comment:
cool. I think the remaining danger here you already realize is if new writable methods are added to SO client, we'll miss them here. Will just have to keep an eye on changes. Maybe another reason to keep the warning around in full